feat: file component support file deletion

soleksy-splunk opened the pull request:
The same PR as https://github.com/splunk/addonfactory-ucc-base-ui/pull/439.

artemrys commented:
@soleksy-splunk Are you going to update test cases and docs? And I also owe you an update to ban encrypted and optional.

soleksy-splunk commented:
@artemrys  yes yes i will update tests, just sunk into other ticket, but plan to finish this today, regarding docs are there any extra places where i should update them?

artemrys commented:
> @artemrys yes yes i will update tests, just sunk into other ticket, but plan to finish this today, regarding docs are there any extra places where i should update them?

`docs` folder should be enough.

artemrys reviewed (commented):
@soleksy-splunk Did you test if something like this (on screenshot) works properly?

<img width="780" alt="image" src="https://github.com/splunk/addonfactory-ucc-generator/assets/11015487/48aef620-ac4e-4231-b78f-6bc88363c990">

soleksy-splunk commented on docs/entity/components.md:
```diff
@@ -477,9 +477,10 @@ This feature allows you to upload a single file.
+| fileSupportMessage                                         | string | To display message inside file component.                  | -             |
+| supportedFileTypes<span class="required-asterisk">*</span> | array  | A list of the file types that the user can upload.         | -             |
+| maxFileSize                                                | number | To set the maximum file size in KB that a user can upload. | 500KB         |
+| fileNameToDisplay                                          | string | File name used for displaying previous files while edit    | -             |
```
yea, i treated it like a general documentation about component but now i read it once again and it looks more like documentation that is focused on mapping globalConfig.json into components. Then yea it does not make sense at all to put it there, it will be just confusing

vtsvetkov-splunk commented on ui/src/main/webapp/components/FileInputComponent.tsx:
```diff
@@ -63,14 +92,14 @@ function FileInputComponent(props) {
                     setErrorMsg('');
                     try {
-                        handleChange(field, textDecoder.decode(fileReader.result));
+                        handleChange(field, textDecoder.decode(fileReader.result as ArrayBuffer));
```
please avoid casting, narrow down the type instead. You have union here: `string | ArrayBuffer | null`. Add checks to make sure we have ArrayBuffer, not string or null.

soleksy-splunk replied:
done

vtsvetkov-splunk commented on ui/src/main/webapp/components/FileInputComponent.tsx:
```diff
@@ -40,10 +57,22 @@ function FileInputComponent(props) {
+      then we display error message "file needs to be reuploaded"
+      as there is no access to data inside due to encription
+     */
+    const [errorMsg, setErrorMsg] = useState<JSX.Element | string>(
```
JSX deprecated, use ReactElement from React instead

soleksy-splunk replied:
i changed it into ReactElement

soleksy-splunk commented:
@artemrys  referring to that single tab element, unfortunately there was an issue but i fixed it and right now it should work properly, i moved file name the logic to work for every file that has any value even if it is encoded

![Screenshot 2023-10-13 at 11 35 41](https://github.com/splunk/addonfactory-ucc-generator/assets/143183665/68ffff31-145b-4b65-af22-304f88791efe)

artemrys reviewed (approved):
There are failures with unreleased version of Splunk but it needs to be fixed upstream so ignoring it right now.
